### DISCUSSION TITLE
superspreading: tools for estimating overdispersion in transmission from different data sources

### DISCUSSION THREAD

**Discussion by adamkucharski** [epiverse-trace/epiverse-trace.github.io#44] [Project Ideas]
# _superspreading_: tools for estimating overdispersion in transmission from different data sources

## Description

This package would provide tools to estimate the extent of superspreading (i.e. individual-level variation in transmission) from two commonly available data sources: 1) reconstructed transmission chains, and 2) outbreak cluster sizes. As well as feeding into transmission models, such estimates can be used to [estimate the probability outbreak will become established](https://cmmid.github.io/visualisations/new-outbreak-probability) in a population following initial importation.

For (1), a common approach is to fit a negative binomial distribution to individual level transmission data ([Lloyd Smith et al, 2005](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002993)); such estimates can then be used to identify factors influencing transmission risk (e.g. [whether a case was previously known](https://academic.oup.com/aje/article/188/7/1319/5426492)).

For (2), parameters can be estimated from the size distribution of outbreak clusters ([Blumberg et al, 2013](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002993)), although for a new outbreak, it may be necessary to handle reporting issues and constrain plausible values of R to ensure identifiability (e.g. [Endo et al, 2020](https://wellcomeopenresearch.org/articles/5-67)). These estimation methods can also be extended to multi-group transmission processes (e.g. [with different ages](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004154)).

## Target audience

* typical end-users: anyone working on quantification of epidemiological characteristics (e.g. incubation periods or transmission risk factors)

* potential contributors: groups who have previously published methodology and applications of these estimation approaches

* key collaborators: colleagues within CMMID who've worked on related analysis

## Interoperability

* inputs: a `data.frame` (or `tibble`) with linked cases and contacts, or perhaps an `epicontacts` object OR a `data.frame` (or `tibble`) with outbreak cluster sizes (which could perhaps also be derived from a line list under some clustering criterion in space and time, e.g. using [_o2geosocial_](https://cran.r-project.org/web/packages/o2geosocial/index.html) )

* outputs: overdispersion estimate with uncertainty

* imports:
    * _epiparameter_
    * _epicontacts_

* used by:
    * _scenario_ (Epiverse package)
    * [_ringbp_](https://github.com/epiforecasts/ringbp)

## Usage

The code below illustrates a typical use of the package, using fictitious code and outputs if needed:

```r
library(superspreading)

# Use case 1 (individual level data): 
transmission_chain_data |>
  fit_offspring_distribution() |>
  estimate_risk_factors() |>
  summarise_results() 

# Use case 2 (cluster data): 
cluster_size_data  |>
  fit_to_clusters()  |>
  summarise_results() 

```

**Comment by sbfnk** [comment]
Nice. Use case 2 (and to some degree use case 1) could potentially use some of the existing functionality in [bpmodels](https://github.com/sbfnk/bpmodels) (see example in the [short vignette](https://github.com/sbfnk/bpmodels/blob/master/vignettes/introduction.md)) - which can 1) give the cluster likelihood which, wrapped into e.g. an optimisation / MCMC algorithm could be used for parameter estimation, 2) simulate transmission trees (though I imagine other/better options might be available for this).

**Comment by adamkucharski** [comment]
Thanks, looks like some useful functions to build on in that package. Given 2 parameters, could derive MLE estimate and CI via profile likelihood as starting point– although may want to think about Bayesian approach in later versions so can incorporate priors from past outbreaks. Simulation of transmission trees is a useful feature to have, as well as the analytical probability of extinction etc. Although adapting `ringbp` to incorporate new estimates might deliver more value for users who want to look at scenarios?

**Comment by joshwlambert** [comment]
Thanks @adamkucharski for opening this discussion and @sbfnk for discussion points.

The development of [{superspreading}](https://github.com/epiverse-trace/superspreading) has now started and is taking place in the [epiverse-trace organisation](https://github.com/epiverse-trace). 

The outline of the package description, functionality and target audience outlined above has been converted into issues in the {superspreading} repository and this will be the focus of the package's development in the short term. If you have new ideas or issues for this project please open an issue on the Github repository rather than continuing the discussion on this thread. This will help keep discussions centralised.

After discussion across developers of {superspreading}, {bpmodels} and {epichains} it appears that the overlapping functionality and complementarity of {superspreading} with {bpmodels} (no longer actively developed) and {epichains} (actively developed) mean that a merging of the two packages may provide a simpler, more intuitive and more comprehensive package to provide to users. Therefore the long term development plan for {superspreading} is for it's functions and documentation to be transplanted into {epichains} (i.e. superspreading will become archived and functions and vignettes will be available from {epichains}). In the immediate future we will develop {superspreading} and {epichains} separately as they are both experimental packages with separate maintainers. If during development that clear divergence arises between the packages, we will reconsider merging them.
